### PR TITLE
Fix for multiple instance deployment in same AWS account

### PR DESCRIPTION
**Pull request by akravchuk1**
Hardcoded resource names collide when more then single instance being deployed in the same AWS account

### CODE REVIEW

File: rds.tf [hunk 1]
```diff
@@ -56,13 +56,14 @@ resource "aws_db_instance" "beekeeper" {
   engine                    = "mysql"
   engine_version            = var.rds_engine_version
   instance_class            = var.rds_instance_class
-  name                      = "beekeeper"
+  // only alphanumeric char allowed in DB name, remove -
+  name                      = "${replace(local.instance_alias, "-", "")}"
```
[the remark this answers is not in the document]
Comment (akravchuk1) [reply rds.tf:60]: I was getting this error so I think only alphanumeric is allowed

```
17:30:00 |   | Error: Error creating DB Instance: InvalidParameterValue: DBName must begin with a letter and contain only alphanumeric characters.
-- | -- | --
17:30:00 |   | status code: 400, request id: af0bb6c0-589a-4d35-8b28-0acde71bc25e, {
17:30:00 |   | AllocatedStorage: 10,
17:30:00 |   | AutoMinorVersionUpgrade: true,
17:30:00 |   | BackupRetentionPeriod: 10,
17:30:00 |   | CopyTagsToSnapshot: false,
17:30:00 |   | DBInstanceClass: "db.t3.small",
17:30:00 |   | DBInstanceIdentifier: "beekeeper-perf",
17:30:00 |   | DBName: "beekeeper-perf",

```

[the remark this answers is not in the document]
Comment (barnharts4) [reply rds.tf:60]: Ah, the restriction is from the underlying DB engine: https://github.com/terraform-providers/terraform-provider-aws/issues/9655

Ok, so nothing you can do here.

File: rds.tf [hunk 2]
```diff
@@ -5,7 +5,7 @@
  */
 
 resource "aws_db_subnet_group" "beekeeper_db_subnet_group" {
-  name        = "beekeeper-db-subnet-group"
+  name        = "${local.instance_alias}-db-subnet-group"
   subnet_ids  = var.rds_subnets
   description = "Beekeeper DB Subnet Group"
```
[the remark this answers is not in the document]
Comment (akravchuk1) [reply rds.tf:10]: done

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,10 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [2.1.3] - 2020-08-05
```
[the remark this answers is not in the document]
Comment (akravchuk1) [reply CHANGELOG.md:6]: Changing for today's date, will change again if needed

**Comment by akravchuk1** [comment]
> looks like TravisCI is failing - unapproving. :)

yeah, noticed that - should be fixed now

**Comment by akravchuk1** [comment]
> I am seeing some other places that probably need instance_name added:
> 
> * https://github.com/ExpediaGroup/apiary-lifecycle/blob/master/lambda.tf#L11
> * check every usage of `var.k8s_app_name` - it defaults to `beekeeper` and I don't see the `instance_alias` being added to it.  If users of this module don't set `var.k8s_app_name`, multiple instances will clash.

Good catch for lambda.tf, I missed it.

**Comment by abhimanyugupta07** [comment]
> Hardcoded resource names collide when more then single instance being deployed in the same AWS account

Why do we need to deploy two instances of beekeeper in same account ? Won't it cause issues with multiple instance deleting data from same location ?

**Comment by barnharts4** [comment]
@abhimanyugupta07 - same AWS account, different datalakes.  See the confluence link I will slack you for our use-case.

**Pull request merged**